### PR TITLE
etcdserver: permission checking of Txn() in authApplierV3

### PR DESCRIPTION
**Pull request by mitake**
/cc @heyitsanthony @xiang90 

I'll work on e2e test for the checking mechanism soon.

### CODE REVIEW

**Comment by xiang90** [comment]
LGTM

defer to @heyitsanthony

**Comment by heyitsanthony** [comment]
lgtm

File: etcdserver/apply_auth.go [hunk 1]
```diff
@@ -61,6 +61,59 @@ func (aa *authApplierV3) DeleteRange(txnID int64, r *pb.DeleteRangeRequest) (*pb
...
+
+		case *pb.RequestOp_RequestDeleteRange:
+			if tv.RequestDeleteRange == nil {
+				continue
+			}
+
+			if !aa.as.IsDeleteRangePermitted(aa.user, tv.RequestDeleteRange.Key, tv.RequestDeleteRange.RangeEnd) {
+				return false
+			}
+
```
Comment (xiang90) [inline etcdserver/apply_auth.go]: remove extra empty line here.

Comment (mitake) [inline etcdserver/apply_auth.go]: ah, sorry for that

**Comment by mitake** [comment]
@xiang90 removed the needless empty line, PTAL

**Comment by xiang90** [comment]
@mitake lgtm. You can merge it when CI turns green. Thanks a lot!

**Comment by mitake** [comment]
@xiang90 thanks, merging it

**Pull request merged**